fix(docutils): try python3 if python not in PATH

boneskull opened the pull request:
I discovered this while using WSL (Ubuntu).  `apt install python` installs `python2.7` (sigh..). To get v3 you need to `apt install python3` (and also `apt install python3-pip`).   The result is an executable named `python3`; not `python`.

This changes the `init` command to use `which` to find `python3`/`python`, as well as the validator.  `python3` is actually _preferred_, because while `python` could be python 2.x, `python3` is most certainly not.

**UPDATE**:

More changes here.

Instead of looking for `pip`, `mike` and `mkdocs` in the `PATH`, I've changed it to use `/path/to/python3 -m <module>` , because not all systems actually put these executables in the `PATH`.  If you're using, say, Homebrew on Mac, these _are_ automatically added to the `PATH`, but on e.g. Ubuntu they are not.

Furthermore, added the ability to provide a custom path to the `typedoc` executable.  This is unneeded in most cases, but can aid development at minimum.

boneskull committed:
fix(docutils): try python3 if python not in PATH

boneskull commented:
Current dependencies on/for this PR:
* boneskull/support-which-types
  * **PR #18650** <a href="https://app.graphite.dev/github/pr/appium/appium/18650" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18650?utm_source=stack-comment).

mykola-mokhnach commented:
This is unlikely to happen, but I would still prefer to point the script explicitly to python3 by default and not just python, which could be an alias to python2

mykola-mokhnach commented:
also macos has only python3 now

boneskull commented:
@mykola-mokhnach Yes, that's what it does--it looks for `python3` first.  The validator will also explicitly check the version, regardless of the name of the executable.

boneskull committed:
fix(docutils): provide CJS executable for proper linking

boneskull committed:
fix(docutils): run mkdocs via python -m

boneskull committed:
fix(docutils): add @types/which as it is exported

boneskull committed:
fix(docutils): exec mike/mkdocs via python -m

- also adds a "custom typedoc path" option
- consolidate the `whichPython*` functions into a single `findPython` fn

boneskull commented:
hmm, yes, but we'd still need to check the output of `python --version`.

mykola-mokhnach commented:
> hmm, yes, but we'd still need to check the output of `python --version`.

you mean to make sure the minimum version requirement is satisfied?

boneskull committed:
chore(docutils): lint

- consolidate some eslint overrides

boneskull commented:
@mykola-mokhnach Yeah, because on my mac using homebrew, python 3 is `python`, not `python3`. on linux, `python` is python 2.  so we don't know what `python` is.
